### ISSUE TITLE
PERFORMANCE, Computational costs, Background process

### ISSUE THREAD

**Issue by rolandsin** [google-research/google-research#1010]
Dear machine learning Aces, 

What happens when I set the epochs setting to 1 ? 
I have been trying to execute the electricity model with the script_hyperparam_opt. 

I know I am not using a GPU, but it takes foreverer... However, my PC parameters are not too bad too...
![image](https://user-images.githubusercontent.com/100216305/156133314-5ce735a5-1cc0-41d4-81d4-c412bbdcaa2e.png)

I really dont get what is happening in the background, when it says and runs the same multiple times:

None
*** Fitting TemporalFusionTransformer ***
Getting batched_data
Using cached training data
Using cached validation data
Using keras standard fit
Train on 450000 samples, validate on 50000 samples
273152/450000 [=================>............] - ETA: 3:30:42 - loss: 0.4841
